### ISSUE TITLE
Lots of <UNK> in real application

### ISSUE THREAD

**Issue by tim5go** [leopard1/BiGRU-Summarization#16]
I found that the algorithm perform very well on the test sets, but very poorly (many <UNK> )if I just pick some random articles online. 

I believe the root problem is that the coverage of the dictionary is not good enough.
Maybe you could consider building the dictionary from a larger dataset ^^

**Comment by leix28** [comment]
I believe 20k words is a relatively big dictionary. Usually, 8k vocabularies are good enough for a English learner to understand a news. 

The model is limited to generate a short summary (about 10 words) from a few sentences (about 30 words in total). If the input is too long, I don't know what the model can generate. Also please normalize the input(lowercase and segmentation). ```Hello word!```->```hello world !```

I admit that this system is far from acceptable for a practical application.

**Comment by tim5go** [comment]
Thanks for your good explanation~
I still have few questions:

1) In order to adjust to a paragraph with ~ 100 words, any parameters in neural network I need to adjust also?

2) I notice in R Nallapati's paper (https://arxiv.org/abs/1602.06023) they implement a Switching Generator-Pointer to handle OOV, is it something that would improve the situation?

**Comment by leix28** [comment]
1. No. But the training data is around 30 words. So it shouldn't work on longer text unless you train with other data.

2. Yes. I would recommend [PointerNet](https://github.com/abisee/pointer-generator) using similar idea but different model. Also, this model is trained on longer documents.
